### PR TITLE
feat: convert cookie to node test runner

### PR DESCRIPTION
**Pull request by ilteoood**
## This relates to...

#2267

## Rationale

Port cookie tests to node test runner

## Changes

Remove tap and use node test runner for the cookie suite

### Features

N/A

### Bug Fixes

N/A

### Breaking Changes and Deprecations

None

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [S] Benchmarked (**optional**)
- [S] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by mcollina** [comment]
you can use https://www.npmjs.com/package/borp to support windows

File: package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "lint": "standard | snazzy",
     "lint:fix": "standard --fix | snazzy",
     "test": "node scripts/generate-pem && npm run test:tap && npm run test:node-fetch && npm run test:fetch && npm run test:cookies && npm run test:wpt && npm run test:websocket && npm run test:jest && npm run test:typescript",
-    "test:cookies": "node scripts/verifyVersion 16 || tap test/cookie/*.js",
+    "test:cookies": "node scripts/verifyVersion 16 || node --test test/cookie/*.js",
```
Comment (pmarchini) [inline package.json:79]: Hey, I noticed that using `node --test` causes nyc to get stuck.   
However, removing the `--test` flag appears to resolve the issue. 

```suggestion
    "test:cookies": "node scripts/verifyVersion 16 || node test/cookie/*.js",
```

Comment (pmarchini) [reply package.json:79]: `"test:cookies": "node scripts/verifyVersion 16 || borp --coverage test/cookie/*.js",` using `borp`

Comment (ilteoood) [reply package.json]: As of now, I would avoid the usage of the `--coverage` parameter, in order to avoid low coverage errors during the migration. wdyt?

**Comment by ilteoood** [comment]
Closing this since it has been merged in another PR

**Pull request closed without merging**